### DISCUSSION TITLE
Introducing the eetasks vscode extension

### DISCUSSION THREAD

**Discussion by lopezvoliver** [gee-community/gee-community.github.io#8] [Show and tell]
Hi, I am working on a [vscode extension](https://marketplace.visualstudio.com/vscode) with the goal of displaying Earth Engine tasks. I have a working version now ([link to repository](https://github.com/lopezvoliver/eetasks)).

Here's a preview of how it looks:

![preview](https://raw.githubusercontent.com/lopezvoliver/eetasks/main/docs/assets/eetasks-readme.gif)

Currently, there are a couple of options to open a new "Tasks view": using a service account (shown above), using the credentials stored by the `python earthengine-api`, or an account authenticated with [gcloud](https://cloud.google.com/sdk/docs/install) (e.g. using the [earthengine cli](https://developers.google.com/earth-engine/guides/command_line). 

This first version works in Linux and Windows, but of course it needs to be tested by more than one user. I am also thinking about adding some additional features (e.g. filter tasks, batch cancel tasks, etc.) in later releases.

You can test the extension after manually installing it using the [Extensions: Install from VSIX](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix). You can download it from the repository [here](https://github.com/lopezvoliver/eetasks/releases/download/v0.0.1/eetasks-0.0.1.vsix). 

Prior to [publishing](https://code.visualstudio.com/api/working-with-extensions/publishing-extension) it to the [VS Code Extension Marketplace](https://marketplace.visualstudio.com/vscode), I would like to know your opinion, and ultimately publish it perhaps under a "gee-community" organization, which would involve [creating an azure organization](https://learn.microsoft.com/azure/devops/organizations/accounts/create-organization) if it doesn't already exist.

**Comment by giswqs** [comment]
This is great! I guess @12rambau would be interested in this as well. See https://github.com/gee-community/ipygee/issues/2#issuecomment-1763032543

**Reply by 12rambau** [reply]
Thanks for the ping, I'm very much interested by this feature it's great to see what you've done for VS code ! 

And effectively I will code a similar widget for direct use in notebooks.

**Reply by lopezvoliver** [reply]
@giswqs  @12rambau , would you want to go ahead and add me to the organization? I'll be glad to add the repository and continue contributing to it. I'd be also interested in contributing to the ipygee effort.

**Reply by giswqs** [reply]
Done. Check email to accept the invite.

**Reply by lopezvoliver** [reply]
Thanks @giswqs  I added and updated the repository. I also created an azure organization (and publisher) with the name gee-community (https://dev.azure.com/gee-community), as a pre-requisite to publish the vscode extension in the Marketplace. I'll be happy to add you if needed. 

You can now find the extension in vscode

https://marketplace.visualstudio.com/items?itemName=gee-community.eetasks

**Reply by 12rambau** [reply]
Sorry @lopezvoliver I don't have any power here but @giswqs was fast enough!
Could you add me to the azure org as well? I'm curious about how it works ;-)

**Reply by lopezvoliver** [reply]
@12rambau Sure, I added you to the organization through your github username. I think you should be able to login with your github account https://dev.azure.com/gee-community/ @giswqs also added you.

**Reply by 12rambau** [reply]
thanks it works like a charm

**Comment by lopezvoliver** [comment]
Dear `gee-community`, allow me to announce here the [new release](https://github.com/gee-community/eetasks/releases/tag/v0.1.0) of the [EE Tasks vscode extension](https://marketplace.visualstudio.com/items?itemName=gee-community.eetasks). 

Here's a preview of an exciting new feature, the `EE Tasks: run GEE script` command:

![image](https://raw.githubusercontent.com/gee-community/eetasks/main/docs/assets/geerunExample.gif) 

[Read more about what can be done (and more importantly, what cannot be done) with this here.](https://github.com/gee-community/eetasks/blob/main/docs/runGEEscripts.md)

Please try it out and let me know what you think.
